packages/contracts/evm-contracts/contracts/AnnotatedMintNft.sol

```diff
@@ -0,0 +1,181 @@
+
+    /// @dev Emitted when `newMinter` is added to the mapping of allowed `minters`.
+    event SetMinter(address indexed newMinter);
+
+    /// @dev Emitted when `oldMinter` is removed from the mapping of allowed `minters`
+    event RemoveMinter(address indexed oldMinter);
+
+    /// @dev Emitted when `baseUri` is updated from `oldUri` to `newUri`.
+    event SetBaseURI(string oldUri, string newUri);
+
+    /// @dev Emitted when a new token with ID `tokenId` is minted, with `initialData` provided in the `mint` function parameters.
+    event Minted(uint256 indexed tokenId, string initialData);
+
+    /// @dev Sets the NFT's `name`, `symbol`, `maxSupply` to `supply`, and transfers ownership to `owner`.
+    /// Also sets `currentTokenId` to 1 and `baseExtension` to `".json"`.
+    /// @param name Collection name.
+    /// @param symbol Collection symbol.
+    /// @param owner The owner of the contract, who will be able to execute
+    constructor(
+        string memory name,
+        string memory symbol,
+        uint256 supply,
+        address owner
+    ) ERC721(name, symbol) {
+        maxSupply = supply;
+        currentTokenId = 1;
+        baseExtension = ".json";
+        transferOwnership(owner);
+    }
+
+    /// @dev Returns true if this contract implements the interface defined by `interfaceID`. See EIP165.
+    function supportsInterface(
+        bytes4 interfaceID
+    ) public pure override(ERC165, ERC721) returns (bool) {
+        return
+            interfaceID == this.supportsInterface.selector || // ERC165
+            interfaceID == this.mint.selector; // ERC721 Paima-extended
+    }
+
+    /// @dev Mints a new token to address `_to`, passing `initialData` to be emitted in the event.
+    /// Increases the `totalSupply` and `currentTokenId`.
+    /// Reverts if `totalSupply` is not less than `maxSupply` or if `_to` is a zero address.
+    /// Emits the `Minted` event.
+    function mint(address _to, string calldata initialData) external canMint returns (uint256) {
+        require(maxSupply > totalSupply, "AnnotatedMintNft: max supply reached");
+        require(_to != address(0), "AnnotatedMintNft: zero receiver address");
+
+        uint256 tokenId = currentTokenId;
+        _safeMint(_to, tokenId);
+
+        totalSupply++;
+        currentTokenId++;
+
+        emit Minted(tokenId, initialData);
+        return tokenId;
+    }
+
+    /// @dev Burns token of ID `_tokenId`. Callable only by the owner of the specified token.
+    /// Reverts if `_tokenId` is not existing.
+    function burn(
+        uint256 _tokenId
+    ) external onlyExistingTokenId(_tokenId) onlyTokenOwner(_tokenId) {
+        totalSupply--;
+        _burn(_tokenId);
+    }
+
+    /// @dev Adds `_minter` to the mapping of allowed `minters` of this NFT.
+    /// Callable only by the contract owner.
+    /// Emits the `SetMinter` event.
+    function setMinter(address _minter) external onlyOwner {
```

Comment:
I realized today that the `AnnotatedMintNft` cannot be used by default (the `canMint` modifier will throw`), so to use it you NEED to set at least 1 minter (or be the contract owner). My guess is that the reason for this is to disable minting unless it's done through the `NativeNftSale` contract (and it supports multiple minters in case you want to provide multiple ways to mint the NFT)

Probably this design should be mentioned in the docs

Reply:
Added a contract-level comment about this.